### PR TITLE
[IN-9] News implementation

### PR DESCRIPTION
**Pull request by glachezar**

### CODE REVIEW

File: investment-app/TRINV.Api/TRINV.Application/Commands/News/CreateNewsCommand.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        _repository = repository;
+        _unitOfWork = unitOfWork;
+        _userContext = userContext;
+    }
+
+    public async Task<OperationResult<News>> Handle(CreateNewsCommand request, CancellationToken cancellationToken)
+    {
+        var operationResult = new OperationResult<News>();
+
+        if (Ensure.IsArgumentNullOrWhiteSpace(request.Name))
```
[the remark this answers is not in the document]
Comment (glachezar) [reply investment-app/TRINV.Api/TRINV.Application/Commands/News/CreateNewsCommand.cs:43]: Done.

File: investment-app/TRINV.Api/TRINV.Infrastructure/Repositories/Repository.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@ public async Task<IEnumerable<T>> GetAllAsync(CancellationToken cancellationToke
 
     public async Task<T?> GetByIdAsync(int id, CancellationToken cancellationToken)
     {
-        return await _context.Set<T>().FindAsync(id, cancellationToken);
+        return await _context.Set<T>().FindAsync(new object[]{id}, cancellationToken);
```
[the remark this answers is not in the document]
Comment (glachezar) [reply investment-app/TRINV.Api/TRINV.Infrastructure/Repositories/Repository.cs:24]: Done.

File: investment-app/TRINV.Api/TRINV.InvestTrackApplication/Controllers/NewsController.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
+﻿namespace TRINV.InvestTrackApplication.Controllers;
+
+using Application.Commands.News;
+using Domain.Entities;
+using Microsoft.AspNetCore.Mvc;
+using MediatR;
+using Shared.Business.Utilities;
+using Application.Queries.News;
+
+[Route("api/[controller]")]
```
[the remark this answers is not in the document]
Comment (glachezar) [reply investment-app/TRINV.Api/TRINV.InvestTrackApplication/Controllers/NewsController.cs:10]: Done.

File: investment-app/TRINV.Api/TRINV.InvestTrackApplication/Controllers/NewsController.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        _mediator = mediator;
+    }
+
+    [HttpPost("createNews")]
+    public async Task<IActionResult> CreateNews([FromBody] CreateNewsCommand command)
+    {
+        return Ok(await _mediator.Send(command));
+    }
+
+    [HttpPost("deleteNews")]
```
[the remark this answers is not in the document]
Comment (glachezar) [reply investment-app/TRINV.Api/TRINV.InvestTrackApplication/Controllers/NewsController.cs:27]: Done.

File: investment-app/TRINV.Api/TRINV.InvestTrackApplication/Controllers/NewsController.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    public async Task<IActionResult> DeleteNews([FromBody] DeleteNewsCommand command)
+    {
+        return Ok(await _mediator.Send(command));
+    }
+
+    [HttpPost("updateNews")]
```
[the remark this answers is not in the document]
Comment (glachezar) [reply investment-app/TRINV.Api/TRINV.InvestTrackApplication/Controllers/NewsController.cs:33]: Done.

**Review by MihaelPavlov — APPROVED** [review]

**Pull request merged**